### PR TITLE
feature: provide jackson jq extra as an independent module (#68)

### PR DESCRIPTION
**Pull request by Jiehong**

### CODE REVIEW

File: integration-tests/src/test/java/io/quarkiverse/jackson/jq/it/JacksonJqLoaderTest.java [hunk 1]
```diff
@@ -27,6 +27,14 @@ public void functionsTest() {
...
-                .body("$", hasItem("myFunction/2")); // custom function not filtered
+                .body("$", hasItem("myFunction/2")) // custom function not filtered
+                .body("$", hasItem("strftime/1")) // extra function not filtered
+                .body("$", hasItem("strftime/2")) // extra function not filtered
+                .body("$", hasItem("uridecode/0")) // extra function not filtered
+                .body("$", hasItem("uriparse/0")) // extra function not filtered
+                .body("$", hasItem("strptime/1")) // extra function not filtered
+                .body("$", hasItem("strptime/2")) // extra function not filtered
+                .body("$", hasItem("timestamp/0")) // extra function not filtered
+                .body("$", hasItem("uuid4/0")); // extra function not filtered
```
Comment (Jiehong) [inline integration-tests/src/test/java/io/quarkiverse/jackson/jq/it/JacksonJqLoaderTest.java:38]: I didn't create a new module just for testing the extra part, but those functions can only come from the extra jq module, so it should be fine IMO

**Comment by Jiehong** [comment]
Tested locally:

```
$ mvn clean test
...
[INFO] 
[INFO] Results:
[INFO] 
[INFO] Tests run: 4, Failures: 0, Errors: 0, Skipped: 0
[INFO] 
[INFO] ------------------------------------------------------------------------
[INFO] Reactor Summary for Quarkus - Jackson Jq - Parent 1.2.0-SNAPSHOT:
[INFO] 
[INFO] Quarkus - Jackson Jq - Parent ...................... SUCCESS [  0.539 s]
[INFO] Quarkus - Jackson Jq - Runtime ..................... SUCCESS [  1.691 s]
[INFO] Quarkus - Jackson Jq - Runtime - Extra ............. SUCCESS [  0.014 s]
[INFO] Quarkus - Jackson Jq - Deployment .................. SUCCESS [  0.593 s]
[INFO] Quarkus - Jackson Jq - Integration Tests ........... SUCCESS [  4.679 s]
[INFO] ------------------------------------------------------------------------
[INFO] BUILD SUCCESS
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  7.635 s
[INFO] Finished at: 2023-01-09T21:20:17+01:00
[INFO] ------------------------------------------------------------------------
```

**Review by ricardozanini — COMMENTED** [review]
Nice! Have you tried on your local project?

**Review by ricardozanini — APPROVED** [review]

**Review by fjtirado — APPROVED** [review]

**Comment by ricardozanini** [comment]
@Jiehong let me know if you need anything else before since I'm planning to release a new version with this feature.

**Comment by ricardozanini** [comment]
@all-contributors please add @Jiehong for code

**Pull request merged**